lib/engine/step/g_18_al/assign.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative '../assign'
+
+module Engine
+  module Step
+    module G18AL
+      class Assign < Assign
+        def process_assign(action)
+          company = action.entity
+          raise GameError, "#{company.owner.name} owns no trains" if company.owner.trains.empty?
+
+          target = action.target
+          hexes = company.abilities(:assign_hexes)&.hexes
+          reachable = @game.graph.reachable_hexes(company.owner).find { |h, _| h.id == target.id }
+          raise GameError, "#{@game.get_location_name(target.id)} is not reachable" unless reachable
```

Comment:
if you do a merge with master there's a new method now

@game.raise_error

Reply:
Done